src/element.ts

```diff
@@ -97,20 +92,37 @@ export abstract class Element {
   protected registry?: Registry;
 
   /**
-   * Some elements include text.
-   * The `textFont` property contains information required to style the text (i.e., font family, size, weight, and style).
-   * It is undefined by default, and can be set using `setFont(...)` or `resetFont()`.
-   */
-  protected textFont?: Required<FontInfo>;
+    */
+  protected textFont: Required<FontInfo>;
+  protected text = '';
+  protected textMetrics: TextMetrics = {
+    fontBoundingBoxAscent: 0,
+    fontBoundingBoxDescent: 0,
+    actualBoundingBoxAscent: 0,
+    actualBoundingBoxDescent: 0,
+    actualBoundingBoxLeft: 0,
+    actualBoundingBoxRight: 0,
+    width: 0,
+  };
+
+  protected width: number = 0;
+  protected yShift: number = 0;
+  protected xShift: number = 0;
```

Comment:
Should the `xShift` line be on top? I normally think of x, y...

Reply:
done